tensorflow_datasets/image/arc.py

```diff
@@ -0,0 +1,158 @@
+      tfds.core.Version(
+          "3.0.0", "New split API (https://tensorflow.org/datasets/splits)"),
+  ]
+
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        # This is the description that will appear on the datasets page.
+        description=_DESCRIPTION,
+        # tfds.features.FeatureConnectors
+        features=tfds.features.FeaturesDict({
+            # Images are of varying size
+            "task_id": tfds.features.Text(),
+            "train": tfds.features.Sequence({
+                "input": tfds.features.Tensor(shape=(30, 30), dtype=tf.uint8),
+                "input_size": tfds.features.Tensor(shape=(2,), dtype=tf.uint8),
+                "output": tfds.features.Tensor(shape=(30, 30), dtype=tf.uint8),
+                "output_size": tfds.features.Tensor(shape=(2,), dtype=tf.uint8),
+            }),
+            "test": tfds.features.Sequence({
+                "input": tfds.features.Tensor(shape=(30, 30), dtype=tf.uint8),
+                "input_size": tfds.features.Tensor(shape=(2,), dtype=tf.uint8),
+                "output": tfds.features.Tensor(shape=(30, 30), dtype=tf.uint8),
+                "output_size": tfds.features.Tensor(shape=(2,), dtype=tf.uint8),
+            })
+        }),
+        # If there's a common (input, target) tuple from the features,
+        # specify them here. They'll be used if as_supervised=True in
+        # builder.as_dataset.
+        supervised_keys=None,
+        # Homepage of the dataset for documentation
+        homepage=_BASE_URL,
+        citation=_CITATION,
+    )
+
+  def _split_generators(self, dl_manager):
+    """Downloads the data, defines the splits and returns SplitGenerators."""
+
+    # dl_manager is a tfds.download.DownloadManager that can be used to
+    # download and extract URLs
+    extracted_dir = dl_manager.download_and_extract(_DL_RESOURCE)
+    data_dir = os.path.join(extracted_dir, _EXTRACT_SUBDIR, "data")
+    train_dir = os.path.join(data_dir, "training")
+    test_dir = os.path.join(data_dir, "evaluation")
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=10,
+            gen_kwargs={
+                "directory": train_dir,
+            }),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TEST,
+            num_shards=10,
+            gen_kwargs={
+                "directory": test_dir,
+            }),
+    ]
+
+  def _to_image(self, data):
+      img = np.zeros(shape=[30, 30], dtype=np.uint8)
+      height = len(data)
+      width = len(data[0])
+      for index, row in enumerate(data):
+          width = max(width, len(row))
+          # Temporary hack to replace null values with 5 (gray).
+          # See https://github.com/fchollet/ARC/pull/2
+          #                              vvvvvvvvvvvvvvvvvvvvvvvvvvv
+          img[index, :len(row)] = [value if value is not None else 5
+                                   for value in row]
```

Comment:
Could you do this using np arrays ?

Something like: 

img[:height,:width] = data
img = np.where(img, img, 5)

Reply:
Sure